### PR TITLE
chore(core): refactor sign-in routes

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Refactor sign-in routes.

## Linear Issue Reference

LOG-1579

## Testing

Tested locally.

### CODE REVIEW

File: packages/core/src/lib/session.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Context } from 'koa';
+import { InteractionResults, Provider } from 'oidc-provider';
+
+// TODO: change this after frontend is ready.
+// Should combine baseUrl(domain) from database with a 'callback' endpoint.
+export const connectorRedirectUrl = 'https://logto.dev/callback';
```
Comment (gao-sun) [inline packages/core/src/lib/session.ts:6]: looks like this should be a fixed route based on current Logto domain which has nothing related to frontend?

Comment (darcyYe) [reply packages/core/src/lib/session.ts:5]: removed improper comments.

File: packages/core/src/routes/session.ts [hunk 2]
```diff
@@ -54,7 +53,13 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
     koaGuard({ body: object({ username: string(), password: string() }) }),
     async (ctx, next) => {
       const { username, password } = ctx.guard.body;
-      await signInWithUsernameAndPassword(ctx, provider, username, password);
+      assertThat(username && password, 'session.insufficient_info');
+
+      const { id } = await findUserByUsernameAndPassword(username, password);
+      await assignInteractionResults(ctx, provider, { login: { accountId: id } });
+      ctx.userLog.userId = id;
+      ctx.userLog.username = username;
```
Comment (wangsijie) [inline packages/core/src/routes/session.ts:61]: this line need to be moved before `findUserByUsernameAndPassword`, then failed attempt can be logged.

File: packages/core/src/routes/session.ts [hunk 3]
```diff
@@ -67,13 +72,30 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
...
         return next();
       }
 
-      await signInWithPhoneAndPasscode(ctx, provider, { jti, phone, code });
+      await verifyPasscode(jti, PasscodeType.SignIn, code, { phone });
+      const { id } = await findUserByPhone(phone);
+
+      await assignInteractionResults(ctx, provider, { login: { accountId: id } });
+      ctx.userLog.userId = id;
+      ctx.userLog.phone = phone;
```
Comment (wangsijie) [inline packages/core/src/routes/session.ts:97]: same as above

**Review by wangsijie — CHANGES_REQUESTED** [review]
be careful to the order of `ctx.userLog` lines, although we have a project to handle logs.

File: packages/core/src/lib/session.ts (hunk 1 above)
Comment (darcyYe) [reply packages/core/src/lib/session.ts:6]: This is a redirect URI after a social connector authorization process.
It's a front-end path that has not been determined, needs to be changed after the front-end implementation.

**Review by wangsijie — APPROVED** [review]

**Pull request merged**